### ISSUE TITLE
gvim text area doesn't fully occupy window dimensions

### ISSUE THREAD

**Issue by DieSpinne** [vim/vim#9700]
### Steps to reproduce

Using Gvim with Athena GUI,

1. `gvim -u NONE -c 'set gfn=*-terminus-*-*-*-*-16-*'` (Note: If you don't have the terminus font you can also use `*-fixed-*-*-*-*-14-*`)
2. Maximize the gvim window

![2022-02-05-092057_1366x768_scrot](https://user-images.githubusercontent.com/62444221/152642430-3269b45a-f3e0-46b8-9b46-92f54c355b38.png)

3. :vsplit and :q

See how a small gap is left behind at the right edge of the window (marked in red below). 

![2022-02-05-092103_1366x768_scrot](https://user-images.githubusercontent.com/62444221/152642437-4282cd1d-e33b-4aed-bf26-209add0bdab5.png)

The gap goes away if I restore and maximize the window again.

### Expected behaviour

The text area spans the whole width of the screen, leaving no gaps behind where a character could fit.

### Version of Vim

8.2.4298

### Environment

OS: Artix
Terminal: Gvim
$TERM: rxvt-unicode-256color
Shell: Bash

**Comment by brammool** [comment]
Is there a specific reason you use the Athena version?  It's very basic.  Most users will use the GTK version.  Not sure it's worth improving the Athena version.

**Comment by DieSpinne** [comment]
Yes. The main reason is the scrollbar — I would use normal vim if it had one —, which can be customized via X-resources in almost all its aspects. GTK gvim, on the other hand, uses my GTK theme, whose scrollbar is light (doesn't blend well with my dark theme) and too wide.

**Comment by k-takata** [comment]
There was the same issue in Windows and it was fixed in[8.2.1228](https://github.com/vim/vim/commit/203ec7760d255d9f92950c8779ddfc587d7896e0).
If `gui_mch_get_scrollbar_xpadding()` and `gui_mch_get_scrollbar_ypadding()` are implemented, the issue will be fixed.

**Comment by DieSpinne** [comment]
@k-takata Thanks, but my issue is not about the scrollbar not being flushed to the window edge, but rather the gap left behind (marked in red) being larger than it should be, such that it steals away some character cells. In other words, the text area, instead of accommodating x characters, accommodates only x-1 or x-2.

**Comment by k-takata** [comment]
See the screenshots in #5602. It's the same.

**Comment by DieSpinne** [comment]
@k-takata In those screenshots, the gaps are not large enough to fit a character. This is the problem I'm reporting here.

**Comment by k-takata** [comment]
Is this fixed in [8.2.4320](https://github.com/vim/vim/commit/28f1a51bde36e2770dd54c9e2ae69a26cafa5a64)?

**Comment by DieSpinne** [comment]
Nope, it only changes the position between the scrollbar and the gap. The text area size is still miscalculated (in the sense that it does not fill the gap as much as possible).

**Comment by dezza** [comment]
@chrisbra this can be closed, at least until re-evaluated. The code was refactored somewhat in https://github.com/vim/vim/commit/dd40b1af5b5800cab81cb4b6566f028282e74031

Thus this is no longer tracking current state.

**Comment by chrisbra** [comment]
Thanks, closing then
